### DISCUSSION TITLE
Same binary for CPU and GPU

### DISCUSSION THREAD

**Discussion by pkese** [xamarin/TorchSharp#633] [Q&A] [answered]
Is there any chance of having a single binary built for both CPU and GPU:
```xml
  <ItemGroup>
    <PackageReference Include="TorchSharp-cpu" Version="0.96.7" />
    <PackageReference Include="TorchSharp-cuda-linux" Version="0.96.7" />
  </ItemGroup>
```
... and then choose what to use in runtime?

Currently the above breaks complaining that CPU and CUDA dlls don't like each other:  
```Two TorchSharp runtime packages have been referenced (both libtorch-cpu and libtorch-cuda)```

My naive workaround is to have 3 projects:
1. actual ML `code`
2. a `cpu-training` project which refrences `code` and includes `TorchSharp-cpu`
3. a `gpu-training` project which refrences `code` and includes `TorchSharp-cuda-linux`  

but it somehow feels wrong having to switch directories all the time.  
And I also have to build and distribute two versions of inference binaries as well.

**Comment by lostmsu** [comment, marked as answer]
Can't you use `cuda-linux` binary for both purposes?

**Reply by pkese** [reply]
Oh ... 🤦‍♂️ yes I can.
